java-desktop/src/main/java/com/couchbase/todo/TodoApp.java

```diff
@@ -2,31 +2,41 @@
 
 import java.io.IOException;
 
+import com.couchbase.todo.controller.ConfigController;
+
 import javafx.application.Application;
+import javafx.fxml.FXML;
 import javafx.fxml.FXMLLoader;
 import javafx.scene.Parent;
 import javafx.scene.Scene;
 import javafx.stage.Stage;
 
 import com.couchbase.todo.controller.LoginController;
 import com.couchbase.todo.controller.MainController;
+import com.couchbase.todo.model.Config;
 import com.couchbase.todo.model.DB;
 
+
 public class TodoApp extends Application {
+    public static volatile Config config;
```

Comment:
There's one more thing I'd like you to consider:

The config, here, does not belong to the app instance.  If the app instance goes away (I know, I know... it won't) the Config will stick around.

This may not be possible without doing really horrible things.  If it is possible, though, it would be awesome if the config belonged to the app,  I think that means that the app instance would have to be statically reachable (as the Config is now).

Have a look to see if that is worth doing.